### PR TITLE
feat(manager/nuget): extract ContainerBaseImage property

### LINKED ISSUE #26250: Support ContainerBaseImage property in dotnet project file
### Discussed in https://github.com/renovatebot/renovate/discussions/26237

<div type='discussions-op-text'>

<sup>Originally posted by **marcmognol** December 11, 2023</sup>
### Tell us more.

With the latest versions of dotnet, it is now possible to build a container image with dotnet sdk:
https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container?pivots=dotnet-8-0

It is also possible to specify the tag of the base image used by the SDK by specifying the ContainerBaseImage property.

I would like Renovate can detect update of the base Docker image specified in ContainerBaseImage property.

Thanks</div>

### PR DESCRIPTION
**Pull request by marcmognol**
## Changes
Extracts the new ContainerBaseImage property from csproj file.

## Context

- Closes #26250 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [X] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/nuget/__fixtures__/with-container-base-image/with-container-base-image-pinned-digest.csproj
Comment (rarkins) [inline lib/modules/manager/nuget/__fixtures__/with-container-base-image/with-container-base-image-pinned-digest.csproj:1]: Please inline these two csproj files into the .spec.ts tests themselves

File: lib/modules/manager/nuget/extract.ts [hunk 1]
```diff
@@ -46,6 +47,20 @@ function extractDepsFromXml(xmlNode: XmlDocument): NugetPackageDependency[] {
     const child = todo.pop()!;
     const { name, attr } = child;
 
+    if (name === 'ContainerBaseImage') {
+      const dep = getDep(child.val, true);
+
+      const currentDepName = dep.depName ?? '';
```
Comment (rarkins) [inline lib/modules/manager/nuget/extract.ts:53]: Package should be skipped if no depName

File: lib/modules/manager/nuget/extract.ts [hunk 2]
```diff
@@ -46,6 +47,20 @@ function extractDepsFromXml(xmlNode: XmlDocument): NugetPackageDependency[] {
...
+
+      results.push({
+        datasource: NugetDatasource.id,
+        depType: 'docker',
+        depName: currentDepName,
+        currentVersion: dep.currentValue,
```
Comment (rarkins) [inline lib/modules/manager/nuget/extract.ts:59]:
```suggestion
        curentValue: dep.currentValue,
```

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/nuget/extract.ts [hunk 3]
```diff
@@ -46,6 +47,20 @@ function extractDepsFromXml(xmlNode: XmlDocument): NugetPackageDependency[] {
     const child = todo.pop()!;
     const { name, attr } = child;
 
+    if (name === 'ContainerBaseImage') {
+      const dep = getDep(child.val, true);
+
+      if (is.nonEmptyStringAndNotWhitespace(dep.depName)) {
+        results.push({
+          datasource: NugetDatasource.id,
```
Comment (viceice) [inline lib/modules/manager/nuget/extract.ts:55]: wrong datasource, needs to get docker

**Review by viceice — COMMENTED** [review]
don't resolve discussions when they're not done

File: lib/modules/manager/nuget/extract.ts (hunk 3 above)
Comment (marcmognol) [reply lib/modules/manager/nuget/extract.ts]: Now fixed. Thanks

File: lib/modules/manager/nuget/__fixtures__/with-container-base-image/with-container-base-image-pinned-digest.csproj
Comment (marcmognol) [reply lib/modules/manager/nuget/__fixtures__/with-container-base-image/with-container-base-image-pinned-digest.csproj:1]: Reduce the length of the files, to make it inline.

File: lib/modules/manager/nuget/extract.ts [hunk 4]
```diff
@@ -46,6 +48,20 @@ function extractDepsFromXml(xmlNode: XmlDocument): NugetPackageDependency[] {
     const child = todo.pop()!;
     const { name, attr } = child;
 
+    if (name === 'ContainerBaseImage') {
+      const dep = getDep(child.val, true);
+
+      if (is.nonEmptyStringAndNotWhitespace(dep.depName)) {
+        results.push({
+          datasource: DockerDatasource.id,
```
Comment (viceice) [inline lib/modules/manager/nuget/extract.ts:56]: needs also be added to supported datasource in index.ts

File: lib/modules/manager/nuget/__fixtures__/with-container-base-image/with-pinned-digest.csproj [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk.Worker">
```
Comment (viceice) [inline lib/modules/manager/nuget/__fixtures__/with-container-base-image/with-pinned-digest.csproj:1]: inline small fixtures as requested

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/nuget/extract.ts (hunk 4 above)
Comment (marcmognol) [reply lib/modules/manager/nuget/extract.ts:56]: Done.

File: lib/modules/manager/nuget/__fixtures__/with-container-base-image/with-pinned-digest.csproj (hunk 5 above)
Comment (marcmognol) [reply lib/modules/manager/nuget/__fixtures__/with-container-base-image/with-pinned-digest.csproj:1]: Are you asking me to copy pase the content of the two fixtures and paste it in the extract.spec.ts file?

Comment (viceice) [reply lib/modules/manager/nuget/__fixtures__/with-container-base-image/with-pinned-digest.csproj:1]: yes, like already done for other tests

Comment (marcmognol) [reply lib/modules/manager/nuget/__fixtures__/with-container-base-image/with-pinned-digest.csproj]: Is it ok for you ?

File: lib/modules/manager/nuget/extract.spec.ts [hunk 6]
```diff
@@ -67,6 +67,52 @@ describe('modules/manager/nuget/extract', () => {
       expect(res?.deps).toHaveLength(17);
     });
 
+    it('extracts ContainerBaseImage', async () => {
+      const contents = `
```
Comment (viceice) [inline lib/modules/manager/nuget/extract.spec.ts:71]:
```suggestion
      const contents = codeBlock`
```
use that helper

File: lib/modules/manager/nuget/extract.spec.ts [hunk 7]
```diff
@@ -67,6 +67,52 @@ describe('modules/manager/nuget/extract', () => {
...
+            datasource: 'docker',
+            currentValue: '7.0.10',
+          },
+        ],
+        packageFileVersion: '0.1.0',
+      });
+    });
+
+    it('extracts ContainerBaseImage with pinned digest', async () => {
+      const contents = `
```
Comment (viceice) [inline lib/modules/manager/nuget/extract.spec.ts:93]:
```suggestion
      const contents = codeBlock`
```

**Review by viceice — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**